jobs/kola-upgrade.Jenkinsfile

```
@@ -0,0 +1,244 @@
+
+        def remoteSession = ""
+        if (params.ARCH != 'x86_64') {
+            // If we're on mArch and using QEMU then initialize the
+            // session on the remote builder
+            stage("Initialize Remote") {
+                pipeutils.withPodmanRemoteArchBuilder(arch: params.ARCH) {
+                    remoteSession = shwrapCapture("""
+                    cosa remote-session create --image ${params.COREOS_ASSEMBLER_IMAGE} --expiration 4h --workdir ${env.WORKSPACE}
+                    """)
+                }
+            }
+        }
+
+        // Run the remaining code in a remote session if we created one.
+        pipeutils.withOptionalExistingCosaRemoteSession(
+                        arch: params.ARCH, session: remoteSession) {
+            stage('BuildFetch') {
+                def commitopt = ''
+                if (params.SRC_CONFIG_COMMIT != '') {
+                    commitopt = "--commit=${params.SRC_CONFIG_COMMIT}"
+                }
+                def ref = pipeutils.get_source_config_ref_for_stream(pipecfg, params.STREAM)
+                pipeutils.shwrapWithAWSBuildUploadCredentials("""
+                cosa init --force --branch ${ref} ${commitopt} ${pipecfg.source_config.url}
+                cosa buildfetch --artifact=qemu --stream=${start_stream} --build=${start_version} --arch=${params.ARCH}
+                cosa decompress --build=${start_version}
+                """)
+
+                // If no target version was specified the target will be the latest build
+                if (target_version == '') {
+                    target_version = shwrapCapture("cosa shell -- jq -r .builds[0].id builds/builds.json")
+                    currentBuild.description = "[${params.STREAM}][${params.ARCH}] - ${start_version}->${target_version}"
+                }
+            }
+
+            // A few independent tasks that can be run in parallel
+            def parallelruns = [:]
+
+            shwrap("""
+            cosa shell -- tee tmp/target_stream.bu <<EOF
+variant: fcos
+version: 1.0.0
+storage:
+  files:
+    - path: /etc/target_stream
+      mode: 0644
+      contents:
+        inline: |
+          ${params.STREAM}
+EOF
+""")
+
+            def kolaparams = [
+                arch: params.ARCH,
+                build: start_version,
+                cosaDir: env.WORKSPACE,
+                extraArgs: "--tag extended-upgrade --allow-rerun-success --append-butane tmp/target_stream.bu",
+                skipBasicScenarios: true,
+                skipUpgrade: true,
+            ]
+            def k1, k2, k3
+
+            switch(params.ARCH) {
+                case 'x86_64':
+                    k1 = kolaparams.clone()
+                    k1.extraArgs += " --qemu-firmware=uefi"
+                    k1.marker = "uefi"
+                    parallelruns['Kola:UEFI'] = { kola(k1) }
+                    if ((start_version[0..1] as Integer) < 34) {
```

Shouldn't this be

```suggestion
                    if ((start_version[0..1] as Integer) >= 34) {
```

?